packages/odyssey-react-mui/src/labs/AppTile.tsx

```diff
@@ -174,8 +205,10 @@ const AppTile = ({
 
       {(onActionClick || auxiliaryText) && (
         <ActionContainer odysseyDesignTokens={odysseyDesignTokens}>
-          {auxiliaryText && <Subordinate>{auxiliaryText}</Subordinate>}
-          {onActionClick && (
+          {auxiliaryText && !isLoading && (
+            <Subordinate>{auxiliaryText}</Subordinate>
+          )}
+          {onActionClick && !isLoading && (
```

Comment:
Since what we're wanting here is to have these display or not based on the loading state, it should probably go first.

```suggestion
          {!isLoading && auxiliaryText && (
            <Subordinate>{auxiliaryText}</Subordinate>
          )}
          {!isLoading && onActionClick && (
```

But even more than that, I think we should probably move this loading state check higher up so we can encompass the whole component rather than having a bunch of `if-else`s spread throughout.

I say this because having so many conditionals typically makes it harder to follow what's happening to the component at any given time. It also makes it harder to test.

I understand we wanna make sure we don't duplicate shared logic, but I'm wondering if we've also added too many `isLoading` conditionals in the process when it's an either-or.

Reply:
@KevinGhadyani-Okta I mentioned this as well. See [this comment](https://github.com/okta/odyssey/pull/2384#issuecomment-2405198575) about moving to a 'loading component'